pandas/core/indexes/base.py

```diff
@@ -5668,8 +5714,17 @@ def get_indexer_non_unique(
             # Item "IndexEngine" of "Union[IndexEngine, ExtensionEngine]" has
             # no attribute "_extract_level_codes"
             tgt_values = engine._extract_level_codes(target)  # type: ignore[union-attr]
+        if is_extension_array_dtype(tgt_values):
```

Comment:
can you check tgt_values.dtype

Reply:
and actually BaseMaskedArray now that i look closer

Reply:
Done, but I hide the local import behind the extension array check

Reply:
Will have a look into adding a is_masked_dtype check soonish to avoid all the local imports